src/Models/Link.php

```diff
@@ -290,4 +292,23 @@ private function getLinkTypes(): array
 
         return $types;
     }
+
+    public function getDisplayTitle(): string
+    {
+        // If we have a title, we can just bail out without any changes
+        if ($this->Title && $this->getURL()) {
```

Comment:
```suggestion
        if ($this->Title) {
```
We don't care about the URL yet - this is explicitly about the title. If there's a title, display it. If not, use the default title.

The default title might use the URL.

Reply:
Updated